[2025-04-08T09:08:21.692+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-04-08T09:08:21.751+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: formula1-etl-pipeline.extract.extract_and_load_to_staging scheduled__2025-03-30T00:00:00+00:00 [queued]>
[2025-04-08T09:08:21.764+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: formula1-etl-pipeline.extract.extract_and_load_to_staging scheduled__2025-03-30T00:00:00+00:00 [queued]>
[2025-04-08T09:08:21.765+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-04-08T09:08:21.807+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): extract.extract_and_load_to_staging> on 2025-03-30 00:00:00+00:00
[2025-04-08T09:08:21.819+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'formula1-etl-pipeline', 'extract.extract_and_load_to_staging', 'scheduled__2025-03-30T00:00:00+00:00', '--job-id', '4', '--raw', '--subdir', 'DAGS_FOLDER/ETL_dag.py', '--cfg-path', '/tmp/tmpn6t0yq9l']
[2025-04-08T09:08:21.825+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=479) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-04-08T09:08:21.827+0000] {standard_task_runner.py:91} INFO - Job 4: Subtask extract.extract_and_load_to_staging
[2025-04-08T09:08:21.827+0000] {standard_task_runner.py:63} INFO - Started process 495 to run task
[2025-04-08T09:08:21.936+0000] {task_command.py:426} INFO - Running <TaskInstance: formula1-etl-pipeline.extract.extract_and_load_to_staging scheduled__2025-03-30T00:00:00+00:00 [running]> on host bbffae879616
[2025-04-08T09:08:22.114+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='formula1-etl-pipeline' AIRFLOW_CTX_TASK_ID='extract.extract_and_load_to_staging' AIRFLOW_CTX_EXECUTION_DATE='2025-03-30T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-03-30T00:00:00+00:00'
[2025-04-08T09:08:22.123+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-04-08T09:08:24.820+0000] {logging_mixin.py:188} WARNING -   0%|          | 0.00/6.28M [00:00<?, ?B/s]
[2025-04-08T09:08:25.760+0000] {logging_mixin.py:188} WARNING -  16%|#5        | 1.00M/6.28M [00:00<00:04, 1.12MB/s]
[2025-04-08T09:08:25.901+0000] {logging_mixin.py:188} WARNING -  32%|###1      | 2.00M/6.28M [00:01<00:02, 2.23MB/s]
[2025-04-08T09:08:26.006+0000] {logging_mixin.py:188} WARNING -  48%|####7     | 3.00M/6.28M [00:01<00:00, 3.46MB/s]
[2025-04-08T09:08:26.131+0000] {logging_mixin.py:188} WARNING -  64%|######3   | 4.00M/6.28M [00:01<00:00, 4.51MB/s]
[2025-04-08T09:08:26.233+0000] {logging_mixin.py:188} WARNING -  80%|#######9  | 5.00M/6.28M [00:01<00:00, 5.64MB/s]
[2025-04-08T09:08:26.353+0000] {logging_mixin.py:188} WARNING -  96%|#########5| 6.00M/6.28M [00:01<00:00, 6.42MB/s]
[2025-04-08T09:08:26.368+0000] {logging_mixin.py:188} WARNING - 100%|##########| 6.28M/6.28M [00:01<00:00, 4.25MB/s]
[2025-04-08T09:08:27.718+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-04-08T09:08:27.719+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/extract_data.py", line 13, in extract_and_load_to_staging
    dataset_path = kagglehub.dataset_download("rohanrao/formula-1-world-championship-1950-2020")
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/kagglehub/datasets.py", line 35, in dataset_download
    path, _ = registry.dataset_resolver(h, path, force_download=force_download)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/kagglehub/registry.py", line 28, in __call__
    return impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/kagglehub/resolver.py", line 29, in __call__
    path, version = self._resolve(handle, path, force_download=force_download)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/kagglehub/http_resolver.py", line 135, in _resolve
    os.remove(archive_path)
FileNotFoundError: [Errno 2] No such file or directory: '/opt/airflow/dataset/datasets/rohanrao/formula-1-world-championship-1950-2020/24.archive'
[2025-04-08T09:08:27.736+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=formula1-etl-pipeline, task_id=extract.extract_and_load_to_staging, run_id=scheduled__2025-03-30T00:00:00+00:00, execution_date=20250330T000000, start_date=20250408T090821, end_date=20250408T090827
[2025-04-08T09:08:27.751+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 4 for task extract.extract_and_load_to_staging ([Errno 2] No such file or directory: '/opt/airflow/dataset/datasets/rohanrao/formula-1-world-championship-1950-2020/24.archive'; 495)
[2025-04-08T09:08:27.773+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-04-08T09:08:27.802+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-04-08T09:08:27.808+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2025-04-08T09:27:28.350+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-04-08T09:27:28.395+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: formula1-etl-pipeline.extract.extract_and_load_to_staging scheduled__2025-03-30T00:00:00+00:00 [queued]>
[2025-04-08T09:27:28.408+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: formula1-etl-pipeline.extract.extract_and_load_to_staging scheduled__2025-03-30T00:00:00+00:00 [queued]>
[2025-04-08T09:27:28.409+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-04-08T09:27:28.432+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): extract.extract_and_load_to_staging> on 2025-03-30 00:00:00+00:00
[2025-04-08T09:27:28.445+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'formula1-etl-pipeline', 'extract.extract_and_load_to_staging', 'scheduled__2025-03-30T00:00:00+00:00', '--job-id', '4', '--raw', '--subdir', 'DAGS_FOLDER/ETL_dag.py', '--cfg-path', '/tmp/tmpt2i41sl7']
[2025-04-08T09:27:28.449+0000] {standard_task_runner.py:91} INFO - Job 4: Subtask extract.extract_and_load_to_staging
[2025-04-08T09:27:28.451+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=538) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-04-08T09:27:28.452+0000] {standard_task_runner.py:63} INFO - Started process 561 to run task
[2025-04-08T09:27:28.561+0000] {task_command.py:426} INFO - Running <TaskInstance: formula1-etl-pipeline.extract.extract_and_load_to_staging scheduled__2025-03-30T00:00:00+00:00 [running]> on host 1c2d39091a25
[2025-04-08T09:27:28.730+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='formula1-etl-pipeline' AIRFLOW_CTX_TASK_ID='extract.extract_and_load_to_staging' AIRFLOW_CTX_EXECUTION_DATE='2025-03-30T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-03-30T00:00:00+00:00'
[2025-04-08T09:27:28.738+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-04-08T09:27:30.215+0000] {logging_mixin.py:188} INFO - Tệp constructors.csv đã tồn tại tại /opt/***/dataset, không di chuyển.
[2025-04-08T09:27:30.216+0000] {python.py:237} INFO - Done. Returned value was: None
[2025-04-08T09:27:30.217+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-04-08T09:27:30.249+0000] {taskinstance.py:1206} INFO - Marking task as SUCCESS. dag_id=formula1-etl-pipeline, task_id=extract.extract_and_load_to_staging, run_id=scheduled__2025-03-30T00:00:00+00:00, execution_date=20250330T000000, start_date=20250408T092728, end_date=20250408T092730
[2025-04-08T09:27:30.401+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2025-04-08T09:27:30.442+0000] {taskinstance.py:3503} INFO - 6 downstream tasks scheduled from follow-on schedule check
[2025-04-08T09:27:30.459+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2025-04-08T09:59:42.050+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-04-08T09:59:42.101+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: formula1-etl-pipeline.extract.extract_and_load_to_staging scheduled__2025-03-30T00:00:00+00:00 [queued]>
[2025-04-08T09:59:42.116+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: formula1-etl-pipeline.extract.extract_and_load_to_staging scheduled__2025-03-30T00:00:00+00:00 [queued]>
[2025-04-08T09:59:42.116+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-04-08T09:59:42.168+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): extract.extract_and_load_to_staging> on 2025-03-30 00:00:00+00:00
[2025-04-08T09:59:42.196+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=367) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-04-08T09:59:42.202+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'formula1-etl-pipeline', 'extract.extract_and_load_to_staging', 'scheduled__2025-03-30T00:00:00+00:00', '--job-id', '4', '--raw', '--subdir', 'DAGS_FOLDER/ETL_dag.py', '--cfg-path', '/tmp/tmpwpfyg7jz']
[2025-04-08T09:59:42.208+0000] {standard_task_runner.py:91} INFO - Job 4: Subtask extract.extract_and_load_to_staging
[2025-04-08T09:59:42.207+0000] {standard_task_runner.py:63} INFO - Started process 394 to run task
[2025-04-08T09:59:42.333+0000] {task_command.py:426} INFO - Running <TaskInstance: formula1-etl-pipeline.extract.extract_and_load_to_staging scheduled__2025-03-30T00:00:00+00:00 [running]> on host a06a1ce74445
[2025-04-08T09:59:42.550+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='formula1-etl-pipeline' AIRFLOW_CTX_TASK_ID='extract.extract_and_load_to_staging' AIRFLOW_CTX_EXECUTION_DATE='2025-03-30T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-03-30T00:00:00+00:00'
[2025-04-08T09:59:42.552+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-04-08T09:59:45.013+0000] {logging_mixin.py:188} WARNING -   0%|          | 0.00/6.28M [00:00<?, ?B/s]
[2025-04-08T09:59:45.919+0000] {logging_mixin.py:188} WARNING -  16%|#5        | 1.00M/6.28M [00:00<00:04, 1.16MB/s]
[2025-04-08T09:59:46.056+0000] {logging_mixin.py:188} WARNING -  32%|###1      | 2.00M/6.28M [00:01<00:01, 2.31MB/s]
[2025-04-08T09:59:46.199+0000] {logging_mixin.py:188} WARNING -  64%|######3   | 4.00M/6.28M [00:01<00:00, 4.90MB/s]
[2025-04-08T09:59:46.387+0000] {logging_mixin.py:188} WARNING -  96%|#########5| 6.00M/6.28M [00:01<00:00, 6.67MB/s]
[2025-04-08T09:59:46.416+0000] {logging_mixin.py:188} WARNING - 100%|##########| 6.28M/6.28M [00:01<00:00, 4.70MB/s]
[2025-04-08T09:59:47.460+0000] {logging_mixin.py:188} INFO - Đã di chuyển circuits.csv về /opt/***/dataset
[2025-04-08T09:59:47.464+0000] {logging_mixin.py:188} INFO - Tệp constructors.csv đã tồn tại tại /opt/***/dataset, không di chuyển.
[2025-04-08T09:59:47.470+0000] {logging_mixin.py:188} INFO - Đã di chuyển constructor_results.csv về /opt/***/dataset
[2025-04-08T09:59:47.487+0000] {logging_mixin.py:188} INFO - Đã di chuyển constructor_standings.csv về /opt/***/dataset
[2025-04-08T09:59:47.502+0000] {logging_mixin.py:188} INFO - Đã di chuyển drivers.csv về /opt/***/dataset
[2025-04-08T09:59:47.507+0000] {logging_mixin.py:188} INFO - Đã di chuyển driver_standings.csv về /opt/***/dataset
[2025-04-08T09:59:47.512+0000] {logging_mixin.py:188} INFO - Đã di chuyển lap_times.csv về /opt/***/dataset
[2025-04-08T09:59:47.517+0000] {logging_mixin.py:188} INFO - Đã di chuyển pit_stops.csv về /opt/***/dataset
[2025-04-08T09:59:47.522+0000] {logging_mixin.py:188} INFO - Đã di chuyển qualifying.csv về /opt/***/dataset
[2025-04-08T09:59:47.527+0000] {logging_mixin.py:188} INFO - Đã di chuyển races.csv về /opt/***/dataset
[2025-04-08T09:59:47.532+0000] {logging_mixin.py:188} INFO - Đã di chuyển results.csv về /opt/***/dataset
[2025-04-08T09:59:47.539+0000] {logging_mixin.py:188} INFO - Đã di chuyển seasons.csv về /opt/***/dataset
[2025-04-08T09:59:47.544+0000] {logging_mixin.py:188} INFO - Đã di chuyển sprint_results.csv về /opt/***/dataset
[2025-04-08T09:59:47.558+0000] {logging_mixin.py:188} INFO - Đã di chuyển status.csv về /opt/***/dataset
[2025-04-08T09:59:47.559+0000] {python.py:237} INFO - Done. Returned value was: None
[2025-04-08T09:59:47.560+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-04-08T09:59:47.577+0000] {taskinstance.py:1206} INFO - Marking task as SUCCESS. dag_id=formula1-etl-pipeline, task_id=extract.extract_and_load_to_staging, run_id=scheduled__2025-03-30T00:00:00+00:00, execution_date=20250330T000000, start_date=20250408T095942, end_date=20250408T095947
[2025-04-08T09:59:47.643+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2025-04-08T09:59:47.678+0000] {taskinstance.py:3503} INFO - 6 downstream tasks scheduled from follow-on schedule check
[2025-04-08T09:59:47.680+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2025-04-08T10:10:04.648+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-04-08T10:10:04.751+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: formula1-etl-pipeline.extract.extract_and_load_to_staging scheduled__2025-03-30T00:00:00+00:00 [queued]>
[2025-04-08T10:10:04.773+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: formula1-etl-pipeline.extract.extract_and_load_to_staging scheduled__2025-03-30T00:00:00+00:00 [queued]>
[2025-04-08T10:10:04.782+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-04-08T10:10:05.171+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): extract.extract_and_load_to_staging> on 2025-03-30 00:00:00+00:00
[2025-04-08T10:10:05.212+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=360) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-04-08T10:10:05.216+0000] {standard_task_runner.py:63} INFO - Started process 390 to run task
[2025-04-08T10:10:05.224+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'formula1-etl-pipeline', 'extract.extract_and_load_to_staging', 'scheduled__2025-03-30T00:00:00+00:00', '--job-id', '3', '--raw', '--subdir', 'DAGS_FOLDER/ETL_dag.py', '--cfg-path', '/tmp/tmpoquf_3mz']
[2025-04-08T10:10:05.245+0000] {standard_task_runner.py:91} INFO - Job 3: Subtask extract.extract_and_load_to_staging
[2025-04-08T10:10:05.447+0000] {task_command.py:426} INFO - Running <TaskInstance: formula1-etl-pipeline.extract.extract_and_load_to_staging scheduled__2025-03-30T00:00:00+00:00 [running]> on host 38a05d0acc45
[2025-04-08T10:10:05.656+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='formula1-etl-pipeline' AIRFLOW_CTX_TASK_ID='extract.extract_and_load_to_staging' AIRFLOW_CTX_EXECUTION_DATE='2025-03-30T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-03-30T00:00:00+00:00'
[2025-04-08T10:10:05.666+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-04-08T10:10:07.610+0000] {logging_mixin.py:188} WARNING -   0%|          | 0.00/6.28M [00:00<?, ?B/s]
[2025-04-08T10:10:08.543+0000] {logging_mixin.py:188} WARNING -  16%|#5        | 1.00M/6.28M [00:00<00:04, 1.13MB/s]
[2025-04-08T10:10:08.679+0000] {logging_mixin.py:188} WARNING -  32%|###1      | 2.00M/6.28M [00:01<00:01, 2.26MB/s]
[2025-04-08T10:10:08.814+0000] {logging_mixin.py:188} WARNING -  64%|######3   | 4.00M/6.28M [00:01<00:00, 4.87MB/s]
[2025-04-08T10:10:08.902+0000] {logging_mixin.py:188} WARNING - 100%|##########| 6.28M/6.28M [00:01<00:00, 5.10MB/s]
[2025-04-08T10:10:08.912+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-04-08T10:10:08.914+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/extract_data.py", line 13, in extract_and_load_to_staging
    dataset_path = kagglehub.dataset_download("rohanrao/formula-1-world-championship-1950-2020")
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/kagglehub/datasets.py", line 35, in dataset_download
    path, _ = registry.dataset_resolver(h, path, force_download=force_download)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/kagglehub/registry.py", line 28, in __call__
    return impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/kagglehub/resolver.py", line 29, in __call__
    path, version = self._resolve(handle, path, force_download=force_download)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/kagglehub/http_resolver.py", line 132, in _resolve
    _extract_archive(archive_path, out_path)
  File "/home/airflow/.local/lib/python3.12/site-packages/kagglehub/http_resolver.py", line 272, in _extract_archive
    raise ValueError(msg)
ValueError: Unsupported archive type.
[2025-04-08T10:10:08.927+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=formula1-etl-pipeline, task_id=extract.extract_and_load_to_staging, run_id=scheduled__2025-03-30T00:00:00+00:00, execution_date=20250330T000000, start_date=20250408T101004, end_date=20250408T101008
[2025-04-08T10:10:08.943+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 3 for task extract.extract_and_load_to_staging (Unsupported archive type.; 390)
[2025-04-08T10:10:08.981+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-04-08T10:10:09.012+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-04-08T10:10:09.019+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2025-04-08T10:40:29.208+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-04-08T10:40:29.274+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: formula1-etl-pipeline.extract.extract_and_load_to_staging scheduled__2025-03-30T00:00:00+00:00 [queued]>
[2025-04-08T10:40:29.296+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: formula1-etl-pipeline.extract.extract_and_load_to_staging scheduled__2025-03-30T00:00:00+00:00 [queued]>
[2025-04-08T10:40:29.297+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-04-08T10:40:29.333+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): extract.extract_and_load_to_staging> on 2025-03-30 00:00:00+00:00
[2025-04-08T10:40:29.361+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=234) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-04-08T10:40:29.375+0000] {standard_task_runner.py:63} INFO - Started process 252 to run task
[2025-04-08T10:40:29.364+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'formula1-etl-pipeline', 'extract.extract_and_load_to_staging', 'scheduled__2025-03-30T00:00:00+00:00', '--job-id', '3', '--raw', '--subdir', 'DAGS_FOLDER/ETL_dag.py', '--cfg-path', '/tmp/tmpe7rt54bw']
[2025-04-08T10:40:29.380+0000] {standard_task_runner.py:91} INFO - Job 3: Subtask extract.extract_and_load_to_staging
[2025-04-08T10:40:29.515+0000] {task_command.py:426} INFO - Running <TaskInstance: formula1-etl-pipeline.extract.extract_and_load_to_staging scheduled__2025-03-30T00:00:00+00:00 [running]> on host b1ea421720de
[2025-04-08T10:40:29.774+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='formula1-etl-pipeline' AIRFLOW_CTX_TASK_ID='extract.extract_and_load_to_staging' AIRFLOW_CTX_EXECUTION_DATE='2025-03-30T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-03-30T00:00:00+00:00'
[2025-04-08T10:40:29.779+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-04-08T10:40:31.605+0000] {logging_mixin.py:188} WARNING -   0%|          | 0.00/6.28M [00:00<?, ?B/s]
[2025-04-08T10:40:32.569+0000] {logging_mixin.py:188} WARNING -  16%|#5        | 1.00M/6.28M [00:00<00:05, 1.09MB/s]
[2025-04-08T10:40:32.706+0000] {logging_mixin.py:188} WARNING -  32%|###1      | 2.00M/6.28M [00:01<00:02, 2.20MB/s]
[2025-04-08T10:40:32.842+0000] {logging_mixin.py:188} WARNING -  64%|######3   | 4.00M/6.28M [00:01<00:00, 4.75MB/s]
[2025-04-08T10:40:32.928+0000] {logging_mixin.py:188} WARNING - 100%|##########| 6.28M/6.28M [00:01<00:00, 4.98MB/s]
[2025-04-08T10:40:33.203+0000] {python.py:237} INFO - Done. Returned value was: /opt/***/dataset/datasets/rohanrao/formula-1-world-championship-1950-2020/versions/24
[2025-04-08T10:40:33.203+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-04-08T10:40:33.230+0000] {taskinstance.py:1206} INFO - Marking task as SUCCESS. dag_id=formula1-etl-pipeline, task_id=extract.extract_and_load_to_staging, run_id=scheduled__2025-03-30T00:00:00+00:00, execution_date=20250330T000000, start_date=20250408T104029, end_date=20250408T104033
[2025-04-08T10:40:33.281+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2025-04-08T10:40:33.305+0000] {taskinstance.py:3503} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2025-04-08T10:40:33.307+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2025-04-08T10:48:57.069+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-04-08T10:48:57.118+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: formula1-etl-pipeline.extract.extract_and_load_to_staging scheduled__2025-03-30T00:00:00+00:00 [queued]>
[2025-04-08T10:48:57.130+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: formula1-etl-pipeline.extract.extract_and_load_to_staging scheduled__2025-03-30T00:00:00+00:00 [queued]>
[2025-04-08T10:48:57.131+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-04-08T10:48:57.155+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): extract.extract_and_load_to_staging> on 2025-03-30 00:00:00+00:00
[2025-04-08T10:48:57.179+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'formula1-etl-pipeline', 'extract.extract_and_load_to_staging', 'scheduled__2025-03-30T00:00:00+00:00', '--job-id', '4', '--raw', '--subdir', 'DAGS_FOLDER/ETL_dag.py', '--cfg-path', '/tmp/tmpsgeswy2l']
[2025-04-08T10:48:57.183+0000] {standard_task_runner.py:91} INFO - Job 4: Subtask extract.extract_and_load_to_staging
[2025-04-08T10:48:57.187+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=269) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-04-08T10:48:57.190+0000] {standard_task_runner.py:63} INFO - Started process 288 to run task
[2025-04-08T10:48:57.284+0000] {task_command.py:426} INFO - Running <TaskInstance: formula1-etl-pipeline.extract.extract_and_load_to_staging scheduled__2025-03-30T00:00:00+00:00 [running]> on host 4e9e6de312e9
[2025-04-08T10:48:57.413+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='formula1-etl-pipeline' AIRFLOW_CTX_TASK_ID='extract.extract_and_load_to_staging' AIRFLOW_CTX_EXECUTION_DATE='2025-03-30T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-03-30T00:00:00+00:00'
[2025-04-08T10:48:57.415+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-04-08T10:48:59.839+0000] {logging_mixin.py:188} WARNING -   0%|          | 0.00/6.28M [00:00<?, ?B/s]
[2025-04-08T10:49:00.777+0000] {logging_mixin.py:188} WARNING -  16%|#5        | 1.00M/6.28M [00:00<00:04, 1.12MB/s]
[2025-04-08T10:49:00.915+0000] {logging_mixin.py:188} WARNING -  32%|###1      | 2.00M/6.28M [00:01<00:01, 2.24MB/s]
[2025-04-08T10:49:01.069+0000] {logging_mixin.py:188} WARNING -  64%|######3   | 4.00M/6.28M [00:01<00:00, 4.71MB/s]
[2025-04-08T10:49:01.200+0000] {logging_mixin.py:188} WARNING -  96%|#########5| 6.00M/6.28M [00:01<00:00, 7.06MB/s]
[2025-04-08T10:49:01.229+0000] {logging_mixin.py:188} WARNING - 100%|##########| 6.28M/6.28M [00:01<00:00, 4.74MB/s]
[2025-04-08T10:49:01.416+0000] {python.py:237} INFO - Done. Returned value was: /opt/***/dataset/datasets/rohanrao/formula-1-world-championship-1950-2020/versions/24
[2025-04-08T10:49:01.417+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-04-08T10:49:01.453+0000] {taskinstance.py:1206} INFO - Marking task as SUCCESS. dag_id=formula1-etl-pipeline, task_id=extract.extract_and_load_to_staging, run_id=scheduled__2025-03-30T00:00:00+00:00, execution_date=20250330T000000, start_date=20250408T104857, end_date=20250408T104901
[2025-04-08T10:49:01.501+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2025-04-08T10:49:01.546+0000] {taskinstance.py:3503} INFO - 6 downstream tasks scheduled from follow-on schedule check
[2025-04-08T10:49:01.548+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2025-04-15T07:09:00.060+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-04-15T07:09:00.106+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: formula1-etl-pipeline.extract.extract_and_load_to_staging scheduled__2025-03-30T00:00:00+00:00 [queued]>
[2025-04-15T07:09:00.125+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: formula1-etl-pipeline.extract.extract_and_load_to_staging scheduled__2025-03-30T00:00:00+00:00 [queued]>
[2025-04-15T07:09:00.126+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-04-15T07:09:00.149+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): extract.extract_and_load_to_staging> on 2025-03-30 00:00:00+00:00
[2025-04-15T07:09:00.163+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'formula1-etl-pipeline', 'extract.extract_and_load_to_staging', 'scheduled__2025-03-30T00:00:00+00:00', '--job-id', '4', '--raw', '--subdir', 'DAGS_FOLDER/ETL_dag.py', '--cfg-path', '/tmp/tmpkbf8rpyq']
[2025-04-15T07:09:00.167+0000] {standard_task_runner.py:91} INFO - Job 4: Subtask extract.extract_and_load_to_staging
[2025-04-15T07:09:00.171+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=416) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-04-15T07:09:00.172+0000] {standard_task_runner.py:63} INFO - Started process 435 to run task
[2025-04-15T07:09:00.249+0000] {task_command.py:426} INFO - Running <TaskInstance: formula1-etl-pipeline.extract.extract_and_load_to_staging scheduled__2025-03-30T00:00:00+00:00 [running]> on host e24a12ae0bf8
[2025-04-15T07:09:00.376+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='formula1-etl-pipeline' AIRFLOW_CTX_TASK_ID='extract.extract_and_load_to_staging' AIRFLOW_CTX_EXECUTION_DATE='2025-03-30T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-03-30T00:00:00+00:00'
[2025-04-15T07:09:00.378+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-04-15T07:09:03.158+0000] {logging_mixin.py:188} WARNING -   0%|          | 0.00/6.28M [00:00<?, ?B/s]
[2025-04-15T07:09:04.125+0000] {logging_mixin.py:188} WARNING -  16%|#5        | 1.00M/6.28M [00:00<00:05, 1.09MB/s]
[2025-04-15T07:09:04.260+0000] {logging_mixin.py:188} WARNING -  32%|###1      | 2.00M/6.28M [00:01<00:02, 2.20MB/s]
[2025-04-15T07:09:04.382+0000] {logging_mixin.py:188} WARNING -  48%|####7     | 3.00M/6.28M [00:01<00:01, 3.33MB/s]
[2025-04-15T07:09:04.556+0000] {logging_mixin.py:188} WARNING -  64%|######3   | 4.00M/6.28M [00:01<00:00, 4.04MB/s]
[2025-04-15T07:09:04.757+0000] {logging_mixin.py:188} WARNING -  80%|#######9  | 5.00M/6.28M [00:01<00:00, 4.40MB/s]
[2025-04-15T07:09:04.949+0000] {logging_mixin.py:188} WARNING -  96%|#########5| 6.00M/6.28M [00:01<00:00, 4.71MB/s]
[2025-04-15T07:09:04.999+0000] {logging_mixin.py:188} WARNING - 100%|##########| 6.28M/6.28M [00:01<00:00, 3.58MB/s]
[2025-04-15T07:09:05.218+0000] {python.py:237} INFO - Done. Returned value was: /opt/***/dataset/datasets/rohanrao/formula-1-world-championship-1950-2020/versions/24
[2025-04-15T07:09:05.219+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-04-15T07:09:05.261+0000] {taskinstance.py:1206} INFO - Marking task as SUCCESS. dag_id=formula1-etl-pipeline, task_id=extract.extract_and_load_to_staging, run_id=scheduled__2025-03-30T00:00:00+00:00, execution_date=20250330T000000, start_date=20250415T070900, end_date=20250415T070905
[2025-04-15T07:09:05.346+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2025-04-15T07:09:05.394+0000] {taskinstance.py:3503} INFO - 6 downstream tasks scheduled from follow-on schedule check
[2025-04-15T07:09:05.398+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
